guides/plugins/apps/flow-builder/add-custom-flow-triggers-from-app-system.md

```diff
@@ -0,0 +1,248 @@
+    <flow-events>
+        <flow-event>
+            ... # The first trigger
+        </flow-event>
+        <flow-event>
+            ... # The second trigger
+        </flow-event>
+    </flow-events>
+    ...
+</flow-extensions>
+```
+
+{% endcode %}
+
+A single flow trigger would look like this:
+
+```xml
+<flow-event>
+    <name>swag.before.open_the_doors</name>
+    <aware>orderAware</aware>
+</flow-event>
+```
+
+| Key          | Required | Description                                                 |
+|:-------------|:---------|:------------------------------------------------------------|
+| name         | yes      | The technical name of your trigger, unique for all actions. |
+| aware        | no       | Filter actions for your trigger, read more bellow.          |
+
+**aware**
+
+The aware will decide which actions are available for your trigger.
+
+***Example***
+
+If you define the orderAware in your trigger config `<aware>orderAware</aware>` the actions related to Order will be available when the trigger selected.
+
+* action.add.order.tag,
+* action.remove.order.tag,
+* action.generate.document,
+* action.grant.download.access,
+* action.set.order.state,
+* action.add.order.affiliate.and.campaign.code,
+* action.set.order.custom.field,
+* action.stop.flow
+
+If you define the customerAware in your trigger config `<aware>orderAware</aware> the actions related to Customer will be available when the trigger selected.
+
+* action.add.customer.tag
+* action.remove.customer.tag
+* action.change.customer.group
+* action.change.customer.status
+* action.set.customer.custom.field
+* action.add.customer.affiliate.and.campaign.code
+* action.stop.flow
+
+For each value when you define, it'll represent one of the `aware` interfaces from the `core`.
+
+To fulfill the aware, refer to a subset of action triggers aware:
+
+| Value              | Interface                                        |
+|:-------------------|:-------------------------------------------------|
+| customerAware      | Shopware\Core\Framework\Event\CustomerAware      |
+| customerGroupAware | Shopware\Core\Framework\Event\CustomerGroupAware |
+| delayAware         | Shopware\Core\Framework\Event\DelayAware         |
+| mailAware          | Shopware\Core\Framework\Event\MailAware          |
+| orderAware         | Shopware\Core\Framework\Event\OrderAware         |
+| salesChannelAware  | Shopware\Core\Framework\Event\SalesChannelAware  |
+| userAware          | Shopware\Core\Framework\Event\UserAware          |
+
+## Trigger API
+
+We provided an API with the endpoint is `POST: /api/_action/trigger-event/{eventName}` to dispatch the custom trigger when you call the API.
+The app calls the API to trigger the custom event and can provide the data. The API will create a CustomAppEvent object and dispatch it with the data provided.
+The data given will be saved through `StorableFlow`. This can be utilized for actions or email templates.
```

Comment:
```suggestion
We provided an API with the endpoint `POST: /api/_action/trigger-event/{eventName}` to dispatch the custom trigger when you call the API.
The app calls the API to trigger the custom event and can provide the data. The API will create a CustomAppEvent object and dispatch it with the data provided.
The data given will be saved through `StorableFlow`. This can be utilized for actions or email templates.
```